### PR TITLE
Knowledgebase Restructure 

### PR DESCRIPTION
**Pull request by sacha-l**
Proposing a slight restructure to make navigating the Knowledgebase from the sidebar more intuitive. Here's an overview of the changes I propose and reasoning: 

- **Renaming "Learn Substrate" to "Substrate Key Concepts".** Makes the parent heading more specific. "Learn Substrate" feels too general for the sub-sections it holds. It also comes off as unclear what the other sections are if they aren't sub-sections of this one. 
- **Taking "Runtime Overview" out of its current section and placing it under "Substrate Key Concepts".** It seems important to present what a Substrate runtime is as a standalone "key concept". To me, the way it's currently placed almost comes to a surprise. What I'm proposing is to merge the two articles (currently "Runtime Overview" and "Runtime Primitives") together so that "Runtime" is its own article and appreciated as a key concept before a new-comer moves on to learning about pallets, FRAME, macros etc. 
- **Removing the "Runtime prefix" in titles of "Runtime" sub-sections.** The repetition feels redundant. With "Runtime" being presented as a "key concept" and renaming the section to "Runtime Principles", my thinking is that it's implied that each articles' context is that of a Runtime. 

kb_restructure

### CODE REVIEW

File: docs/knowledgebase/runtime/index.md [hunk 1]
```diff
@@ -31,11 +64,41 @@ including [over 50 that ship with Substrate](../../knowledgebase/runtime/frame#p
...
+- `AccountId`: The type used to identify user accounts in the runtime.
+
+- `Event`: The type used for events emitted by the runtime.
+
+- `Version`: A type which represents the version of the runtime.
+
+Although a lot of core runtime development can be enabled with FRAME and 
+its related primitives, FRAME is not the only system for developing 
+Substrate based blockchains.
+## Next Steps
```
Comment (jimmychu0807) [inline docs/knowledgebase/runtime/index.md:89]:
```suggestion

## Next Steps
```

File: docs/knowledgebase/runtime/index.md [hunk 2]
```diff
@@ -13,16 +13,49 @@ changes to this state.
 In order to provide its defining forkless runtime upgrade capabilities, Substrate uses runtimes that
 are built as [WebAssembly (Wasm)](../../knowledgebase/getting-started/glossary#webassembly-wasm)
 bytecode. Substrate also defines the
-[core primitives](../../knowledgebase/runtime/primitives#core-primitives) that the runtime must
+core primitives that the runtime must
 implement.
+## Core Primitives
```
Comment (jimmychu0807) [inline docs/knowledgebase/runtime/index.md:18]:
```suggestion

## Core Primitives
```

Comment (jimmychu0807) [reply docs/knowledgebase/runtime/index.md:18]: Prefer adding an empty line before a (sub)title

File: docs/knowledgebase/runtime/index.md [hunk 3]
```diff
@@ -31,11 +64,41 @@ including [over 50 that ship with Substrate](../../knowledgebase/runtime/frame#p
...
+- `Version`: A type which represents the version of the runtime.
+
+Although a lot of core runtime development can be enabled with FRAME and 
+its related primitives, FRAME is not the only system for developing 
+Substrate based blockchains.
+## Next Steps
 
 ### Learn More
 
+- Learn about the [Substrate FRAME](frame).
```
Comment (jimmychu0807) [inline docs/knowledgebase/runtime/index.md:93]: did you define the link `frame`?

File: docs/knowledgebase/runtime/macros.md [hunk 4]
```diff
@@ -603,7 +603,7 @@ public-private key pair from a seed.
 - `Pair` struct type is declared to wrap over the crypto pair. This type implements
   `sp_application_crypto::Pair` and `sp_application_crypto::AppKey` traits determining how it
   generates public-private key pairs from a phrase or seed.
-
+
```
Comment (jimmychu0807) [inline docs/knowledgebase/runtime/macros.md:606]:
```suggestion

```

Comment (jimmychu0807) [reply docs/knowledgebase/runtime/macros.md:606]: We can config our editor to trim spaces at end of lines.

File: website/sidebars.json [hunk 5]
```diff
@@ -7,19 +7,18 @@
...
+      "knowledgebase/learn-substrate/tx-pool",
       "knowledgebase/learn-substrate/session-keys",
       "knowledgebase/learn-substrate/weight",
       "knowledgebase/learn-substrate/off-chain-features"
     ],
-    "Runtime": [
-      "knowledgebase/runtime/index",
-      "knowledgebase/runtime/primitives",
-      "knowledgebase/runtime/frame",
+    "Runtime Principles": [
```
Comment (jimmychu0807) [inline website/sidebars.json:19]:
```suggestion
    "Substrate Runtime": [
```

Comment (jimmychu0807) [reply website/sidebars.json:19]: `Principles` sounds like conceptual and high level. These topics below are somewhat on actual development/implementation level. So I won't be calling this principles.

But feel free to update to other terms if you don't like the name `Substrate Runtime`.

File: website/sidebars.json [hunk 6]
```diff
@@ -7,19 +7,18 @@
       "knowledgebase/getting-started/windows-users",
       "knowledgebase/getting-started/glossary"
     ],
-    "Learn Substrate": [
-      "knowledgebase/learn-substrate/extrinsics",
-      "knowledgebase/learn-substrate/tx-pool",
+    "Substrate Key Concepts": [
+      "knowledgebase/runtime/index",
+      "knowledgebase/learn-substrate/extrinsics",
```
Comment (jimmychu0807) [inline website/sidebars.json:12]:
```suggestion
      "knowledgebase/learn-substrate/extrinsics",
```

Comment (jimmychu0807) [reply website/sidebars.json:12]: trimming space at end of line.

**Review by jimmychu0807 — COMMENTED** [review]
@sacha-l I am mostly happy with the restructure. Just some minor enhancement on spacing and one suggestion on the category name. What do you think?

File: docs/knowledgebase/runtime/index.md [hunk 7]
```diff
@@ -13,16 +13,49 @@ changes to this state.
...
+
+- `Hash`: A type which encodes a cryptographic digest of some data. Typically just a 256-bit
+  quantity.
+
+- `DigestItem`: A type which must be able to encode one of a number of "hard-wired" alternatives
+  relevant to consensus and change-tracking as well as any number of "soft-coded" variants, relevant
+  to specific modules within the runtime.
+
+- `Digest`: A series of DigestItems. This encodes all information that is relevant for a
+  light-client to have on hand within the block.
```
Comment (NukeManDan) [inline docs/knowledgebase/runtime/index.md:34]: Do we define a light client? can we link to it?

File: docs/knowledgebase/runtime/index.md (hunk 3 above)
Comment (NukeManDan) [reply docs/knowledgebase/runtime/index.md:93]:
```suggestion
- Learn about the [Substrate FRAME](./frame).
```
perhaps?

**Review by NukeManDan — APPROVED** [review]
I agree with the changes here with the minor edits Jimmy suggests. 

Will be worth gauging the community if this seems more clear once live.

File: website/sidebars.json (hunk 5 above)
Comment (sacha-l) [reply website/sidebars.json:19]: Thanks for the review @jimmychu0807 . Good point on the name. What about Runtime Development ?

Comment (jimmychu0807) [reply website/sidebars.json:19]: Sure `Runtime Development` sounds good.

**Review by jimmychu0807 — APPROVED** [review]

**Pull request merged**